### DISCUSSION TITLE
Collapsing Frames

### DISCUSSION THREAD

**Discussion by bespinosaaco** [TomSchimansky/CustomTkinter#2608] [Q&A]
Hi, fantastic GUI package.
I'm having issues with Frames collapsing/disappearing when I use certain functions from Tkinter. I'm an amateur programmer learning GUIs with Python as an extra for grad school. Below, there is an example code of the problem I encountered. When I click the button to prompt the import dialogue, but I cancel instead of selecting it and repeat the process, Frame 2 collapses. Commented, I added the tweak I found to go around this problem, but I guess this is not efficient when I have multiple Frames and Tkinter functionalities. 
Is there a more efficient solution? 

```
import customtkinter as ctk
from CTkMessagebox import CTkMessagebox
from tkinter import filedialog

class Frame1(ctk.CTkFrame):
    def __init__(self, master, **kwargs):
        super().__init__(master, **kwargs)
        self.grid_columnconfigure(0, weight=1)
        self.grid_rowconfigure(0, weight=1)
        self.button1 = ctk.CTkButton(self, text="Import CSV", command=self.import_file)
        self.button1.grid(row=2, column=0, padx=10, pady=10, sticky="ew")

    def import_file(self):
        file_path = filedialog.askopenfilename(
            title="Select Statement CSV",
            filetypes=[("CSV Files", "*.csv")]
        )
        if not file_path:
            CTkMessagebox(title="Error", message="Please select a CSV file.", icon='cancel')

            # if hasattr(self.master, 'frame2'):
            #     self.master.frame2.destroy()  # Destroy the old frame2 if it exists

            # self.master.frame2 = Frame2(self.master)  # Recreate Frame2
            # self.master.frame2.grid(row=0, column=1, padx=10, pady=10, sticky="nsew")
            # self.master.frame2.update()
            # self.master.frame2.lift()
            return

class Frame2(ctk.CTkFrame):
    def __init__(self, master, **kwargs):
        super().__init__(master, **kwargs)
        self.grid_rowconfigure(0, weight=1)
        self.grid_columnconfigure(0, weight=1)

class App(ctk.CTk):
    def __init__(self):
        super().__init__()
        ctk.set_appearance_mode('dark')
        self.title('Test')
        self.geometry("700x400")

        self.grid_columnconfigure([0, 1], weight=1)
        self.grid_rowconfigure(0, weight=1)

        self.frame1 = Frame1(self)
        self.frame1.grid(row=0, column=0, padx=10, pady=10, sticky="nsew")
        self.frame1.grid_propagate(False)

        self.frame2 = Frame2(self)
        self.frame2.grid(row=0, column=1, padx=10, pady=10, sticky="nsew")
        self.frame2.grid_propagate(False)

if __name__ == "__main__":
    app = App()
    app.mainloop()
```

**Comment by dipeshSam** [comment]
Please describe your problem more clearly. Also, any visual attachment would be helpful.

**Reply by bespinosaaco** [reply]
Thank you for taking the time to respond.
In the attached screenshots you can see:
-First, the app consisting of two frames with the one on the left having an import button. 
-Second, after clicking the "Import CSV" button, the importing CSV dialogue is prompted as usual.
Here is when I instead of choosing the CSV, click on cancel.
-Third (test3.png), when I click again on the import button Frame2 (the one on the right) collapses along with whatever you have inside that frame.
Did I explain myself? Keep asking if I didn't.
![test1](https://github.com/user-attachments/assets/55e3c762-59b5-49e9-9bff-731d1c331c7c)
![test2](https://github.com/user-attachments/assets/cd6d10ad-9742-48cc-a76a-c7ab313847a2)
![test3](https://github.com/user-attachments/assets/d2537ff1-b511-4157-85c6-708419f48b38)

**Reply by bespinosaaco** [reply]
As I said, the way to solve this that I found was to remake the frame right after is destroyed with the commented code. However, that is not efficient when you have 6 or 7 frames that collapse with more than one button. 
PS: frames only collapse if you don't do what is intended for the button, if you import, everything goes well.

**Comment by dipeshSam** [comment]
I did not find anything wrong with the code you provided and you described it. I tested canceling the box from window and from button but the `Frame2` still remains same without collapsing or changing its size in any way. I also tested inserting CVS test in both the frame, again working well.

Subsequent screenshots:
![Screenshot (226)](https://github.com/user-attachments/assets/ce1d530b-58e7-4e9d-b3de-2b1ff70a7512)
![Screenshot (227)](https://github.com/user-attachments/assets/120ddfb2-dcd6-4f9d-9b3b-54bf6d085fa4)
![Screenshot (228)](https://github.com/user-attachments/assets/36487525-eed2-4235-a8b1-ef4db91f4d34)
![Screenshot (229)](https://github.com/user-attachments/assets/7e2938c3-e6ed-4f84-9f25-325d58375181)
![Screenshot (230)](https://github.com/user-attachments/assets/168565f3-49b5-42a7-b1e8-0addb8f6fecf)


Is there anything else you want to be tested? Or could you please provide a video demonstration of your problem what you are experiencing and expecting?

Here is your updated code I used to place items on second (on the right side) frame:
``` Python 
import customtkinter as ctk
from CTkMessagebox import CTkMessagebox
from tkinter import filedialog

class Frame1(ctk.CTkFrame):
    def __init__(self, master, **kwargs):
        super().__init__(master, **kwargs)
        self.grid_columnconfigure(0, weight=1)
        self.grid_rowconfigure(0, weight=1)
        self.button1 = ctk.CTkButton(self, text="Import CSV", command=self.import_file)
        self.button1.grid(row=2, column=0, padx=10, pady=10, sticky="ew")

    def import_file(self):
        file_path = filedialog.askopenfilename(
            title="Select Statement CSV",
            filetypes=[("CSV Files", "*.csv")]
        )
        if not file_path:
            CTkMessagebox(title="Error", message="Please select a CSV file.", icon='cancel')

            # if hasattr(self.master, 'frame2'):
            #     self.master.frame2.destroy()  # Destroy the old frame2 if it exists

            # self.master.frame2 = Frame2(self.master)  # Recreate Frame2
            # self.master.frame2.grid(row=0, column=1, padx=10, pady=10, sticky="nsew")
            # self.master.frame2.update()
            # self.master.frame2.lift()
            return
        
        else:
            with open(file_path, "r") as file:
                file_data = file.read()          
                columns = file_data.split(",")
                for col in columns:
                    ctk.CTkLabel(self.frame2_ref, text=col, fg_color="darkgreen", corner_radius=20).pack(side="left")

class Frame2(ctk.CTkFrame):
    def __init__(self, master, **kwargs):
        super().__init__(master, **kwargs)
        self.grid_rowconfigure(0, weight=1)
        self.grid_columnconfigure(0, weight=1)
        
class App(ctk.CTk):
    def __init__(self):
        super().__init__()
        ctk.set_appearance_mode('dark')
        self.title('Test')
        self.geometry("700x400")

        self.grid_columnconfigure([0, 1], weight=1)
        self.grid_rowconfigure(0, weight=1)
        
        self.frame1 = Frame1(self)
        self.frame1.grid(row=0, column=0, padx=10, pady=10, sticky="nsew")
        self.frame1.grid_propagate(False)
        
        self.frame2 = Frame2(self)
        self.frame2.grid(row=0, column=1, padx=10, pady=10, sticky="nsew")
        self.frame2.grid_propagate(False)
                
        
        self.frame1.frame2_ref = self.frame2



if __name__ == "__main__":
    app = App()
    app.mainloop()
```


Please feel free to discuss further problems,
Regards.

**Reply by bespinosaaco** [reply]
Thank you so much. So that probably means that my coworker and I are having issues outside of the code. It could be the environment, I don't know. 
Thank you again.

**Reply by dipeshSam** [reply]
You are properly using `frame.grid_propagate(False)` and `sticky="nsew"`, so there must **not** be the case of collapsing the frame. Also, make sure if you're packing sub-widgets then use `pack_propagate(False)`.

If you encounter any further issue regarding this, please fill this thread.

Regards.
